server/src/server/wrapper/connected/CCloudTopics.ts

```diff
@@ -15,12 +17,19 @@ export class CcloudTopics implements Topics {
     }
 
     async createTopic(name: string, partitionCount: number): Promise<void> {
-        await executeCli([
-            "kafka", "topic",
-            "create", name,
-            "--partitions", partitionCount + "",
-            "--cluster", process.env.TIKA_CCLOUD_CLUSTER_ID
-        ]);
+        let topics = await this.getTopics();
+
+        let topicFound = topics.find(v => v.valueOf() === name.valueOf());
+        if (topicFound === undefined) {
```

Comment:
Could we save a call by trying to create the topic and then throwing if the response states that the topic name exits?

Reply:
Sure. It's fair bit more work and some extra parsing, but that could be done.